### ISSUE TITLE
[Suggestion] Custom Guild Prefixes

### ISSUE THREAD

**Issue by KaizerZen** [jagrosh/JDA-Utilities#37]
So... My users recently asked for a way to change the default bot's prefix per guild, and since commands are managed by JDA-Utils it should have a way of setting a custom guild prefix.

**Comment by Shengaero** [comment]
We have been discussing ways to implement a *light* datasystem for doing such a thing.

However, at this moment in time implementation is not scheduled.

**Comment by KaizerZen** [comment]
Okay.

Anyway there's a function on CCB I can use to set a custom implementantion of MessageReceivedEvent to detect a Guild custom prefix?

**Comment by jagrosh** [comment]
There's not but you could either fork the repo and add it, or make a class that extends CommandClientImpl and use that

**Comment by KaizerZen** [comment]
Fine.

**Comment by Shengaero** [comment]
Just to update this guild specific configurations and data management will be a supported feature in 2.0
You can see these changes in the `version-major/2.0`, and even check out the branch using the following build-scripts:

**Gradle**
```groovy
repositories {
    maven { url 'https://jitpack.io' }
}
```

```groovy
dependencies {
    compile 'com.github.JDA-Applications:JDA-Utilities:version-major~2.0-SNAPSHOT'
}
```

**Maven**
```xml
  <repository>
    <id>jitpack.io</id>
    <url>https://jitpack.io</url>
  </repository>
```

```xml
  <dependency>
    <groupId>com.github.JDA-Applications</groupId>
    <artifactId>JDA-Utilities</artifactId>
    <version>version-major~2.0-SNAPSHOT</version>
    <scope>compile</scope>
  </dependency>
```

As a final note, [this gist](https://gist.github.com/TheMonitorLizard/4f09ac2a3c9d8019dc3cde02cc456eee) shows a detailed compilation of all the changes in 2.0, and I highly recommend you read it, as there are breaking changes that occur.

**Comment by Shengaero** [comment]
Addressed in #51
